File: python/reverse_integer.py
class Solution:
    def reverse(self, x: int) -> int:
        '''
		The easiest way to do it is turning number into string and than operate on it.
		'''
        if x >= 0:  # in case there's no need to treat minus
            x = int(str(x)[::-1])  # just turn number into string, reverse it and turn it back into int
        else:
            x = int('-' + str(abs(x))[::-1])  # otherwise get absolute value, do the same and add minus at the beginning

        if (-2) ** 31 <= x <= 2 ** 31 - 1:
            return x
        else:  # in case x is smaller or bigger than required range, return 0
            return 0

File: javascript/reverse_integer.js
var reverse = function(x) {
    /*
    The algorithm:
    - turn into a string
    - turn into an array
    - reverse an array
    - turn back into a string
    - turn back into a Number

    In case of negative number, get absolute value and add minus before conversion to Number.

    Faster syntax was used for if else statements: logicalExpression ? executeWhenTrue : executeWhenFalse
    */
    x = x > 0 ? Number(x.toString().split("").reverse().join("")) : Number('-' + Math.abs(x).toString().split("").reverse().join(""));
    return -(2 ** 31) < x && x < 2 ** 31 ? x : 0;
};